### PR TITLE
[quant][graphmode][fx] Add support for quantizing functional linear + {functional relu/module relu}

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #50976 [quant][graphmode][fx] cleanup linear module test case
* **#50975 [quant][graphmode][fx] Add support for quantizing functional linear + {functional relu/module relu}**

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26032532](https://our.internmc.facebook.com/intern/diff/D26032532)

### CODE REVIEW

File: torch/quantization/fx/quantization_patterns.py [hunk 1]
```diff
@@ -367,19 +367,27 @@ def convert(self, quantizer: QuantizerCls, node: Node, load_arg: Callable,
...
+                    relu_args = [op_out]
+                    relu_args.extend(load_arg(quantized=False)(self.relu_node.args[1:]))
+                    relu_kwargs = load_arg(quantized=False)(self.relu_node.kwargs)
+                    op_out = quantizer.quantized_graph.create_node(
+                        "call_function", torch.nn.functional.relu, tuple(relu_args), relu_kwargs)
+
                 if activation_statically_quantized:
                     # quantize output for statically quantized linear op
                     root_module = quantizer.modules['']
+                    act_post_process_name = self.relu_node.name if self.relu_node else self.linear_node.name
```
Comment (supriyar) [inline torch/quantization/fx/quantization_patterns.py:382]: if debug is true then do we still insert observer at the output of relu node? relu doesn't need output_scale and zero_point, right?

**Review by supriyar — APPROVED** [review]
thanks for the quick fix! left a small comment.

**Comment by supriyar** [comment]
@jerryzh168, do we need similar bugfix for conv2d + relu fusion as well? Seems like we don't do fusion there either in convert.

**Comment by jerryzh168** [comment]
> @jerryzh168, do we need similar bugfix for conv2d + relu fusion as well? Seems like we don't do fusion there either in convert.

yes we do, will add that in a later PR

File: torch/quantization/fx/quantization_patterns.py (hunk 1 above)
Comment (jerryzh168) [reply torch/quantization/fx/quantization_patterns.py:382]: yes, we still insert in the output of relu. debug does not affect where observer is inserted, it only affect the quantized model we produce.
relu itself does not need output_scale/output_zero_point, but linear_relu and conv_relu need these.

**Pull request closed without merging**